packages/react/src/Header/Header.tsx

```diff
@@ -4,61 +4,145 @@
   /** The url for the link on the logo. */
   logoLink?: string
   /** The accessible text for the link on the logo. */
   logoLinkTitle?: string
-  /** A button to toggle the visibility of a Mega Menu. */
-  menu?: ReactNode
+  /** A slot for the menu items. Use Header.MenuLink here. */
+  menuItems?: ReactNode
+  /** The text for the menu button. */
+  menuButtonText?: string
+  /** The accessible label for the navigation section. */
+  navigationLabel?: string
+  /** Whether the menu button is visible on wide screens.  */
+  noMenuButtonOnWideScreen?: boolean
 } & HTMLAttributes<HTMLElement>
 
-export const Header = forwardRef(
+const HeaderRoot = forwardRef(
   (
     {
       appName,
       className,
-      links,
+      children,
       logoBrand = 'amsterdam',
       logoLink = '/',
       logoLinkTitle = 'Ga naar de homepage',
-      menu,
+      menuItems,
+      menuButtonText = 'Menu',
+      navigationLabel = 'Hoofdnavigatie',
+      noMenuButtonOnWideScreen,
       ...restProps
     }: HeaderProps,
     ref: ForwardedRef<HTMLElement>,
   ) => {
+    const [open, setOpen] = useState(false)
+
+    const isWideScreen = useMediaQuery('wide')
+
+    useEffect(() => {
+      // Close the menu when the menu button disappears
+      if (noMenuButtonOnWideScreen && isWideScreen) {
+        setOpen(false)
+      }
+    }, [isWideScreen])
+
     return (
-      <>
-        <header {...restProps} ref={ref} className={clsx('ams-header', className)}>
-          <a href={logoLink} className="ams-header__logo-link">
+      <header {...restProps} ref={ref} className={clsx('ams-header', className)}>
+        <div className="ams-header__brand-section">
+          <a className="ams-header__logo-link" href={logoLink}>
             <span className="ams-visually-hidden">{logoLinkTitle}</span>
             <Logo brand={logoBrand} />
           </a>
-          {links && <div className="ams-header__links">{links}</div>}
-          {menu && <div className="ams-header__menu">{menu}</div>}
           {appName && (
-            <div className="ams-header__app-name">
-              <Heading level={1} size="level-3" className="ams-header__app-name-heading">
-                {appName}
-              </Heading>
-            </div>
+            <Heading level={1} size="level-5" className="ams-header__app-name">
+              {appName}
+            </Heading>
           )}
-        </header>
-      </>
+        </div>
+        {(children || menuItems) && (
+          <nav className="ams-header__navigation" aria-labelledby="primary-navigation">
+            <h2 id="primary-navigation" className="ams-visually-hidden">
+              {navigationLabel}
+            </h2>
+
+            {/* The logo and app name section is recreated here, to make sure the page menu breaks at the right spot */}
+            <div className="ams-header__brand-section ams-header__brand-section--hidden" aria-hidden>
+              <div className="ams-header__logo-link">
```

Comment:
Why do you have the logo a second time?

Reply:
I used a hack to be able to have all header navigation in 1 `nav`. The `nav` contains a recreation of the logo and app name which is placed behind the real logo and app name. This way, you have the correct semantics and you keep the wrapping behaviour as designed.